### PR TITLE
Sc 196416/config map service

### PR DESCRIPTION
**Pull request by russelldeguzman**
Cbor config service messages

### CODE REVIEW

File: config_cbor_map_srv_reply_msg.cpp [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+      err = CborErrorIllegalType;
+      printf("expected string key but got something else\n");
+      break;
+    }
+    err = cbor_value_advance(&value);
+    if (err != CborNoError) {
+      break;
+    }
+    size_t buflen = d.cbor_encoded_map_len;
+    uint8_t *buf = static_cast<uint8_t *>(pvPortMalloc(buflen));
```
[the remark this answers is not in the document]
Comment (victorsowa12) [reply config_cbor_map_srv_reply_msg.cpp:249]: If the caller has to free it, can we have the caller allocate it and pass it in?

[the remark this answers is not in the document]
Comment (victorsowa12) [reply config_cbor_map_srv_reply_msg.cpp:249]: I see that for the sys_info_svc_repl_msg we also have a pvPortMalloc

**Review by victorsowa12 — APPROVED** [review]
Got a question on using FreeRTOS in bm_common_messages, but otherwise looks good!

File: config_cbor_map_srv_reply_msg.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (russelldeguzman) [reply config_cbor_map_srv_reply_msg.cpp:249]: > not necessarily a problem rn, but if this is in bm_common_messages, will someone be able to port this into their own project using say Zephyr?

Nope, definitely not! Punting!

> If the caller has to free it, can we have the caller allocate it and pass it in?

The caller doesn't know how large to make the buffer.

[the remark this answers is not in the document]
Comment (towynlin) [reply config_cbor_map_srv_reply_msg.cpp:249]: Yeah, great question — I love that you're thinking about this. I would certainly rather keep it portable if there's a good design feasible, but sometimes it's not worth it. Related discussion on one of my PRs to bm_protocol recently: https://github.com/bristlemouth/bm_protocol/pull/15#pullrequestreview-1688805984

I also just did a search in the bristlmouth org on github for pvPortMalloc, and there are already plenty of references. The day someone is interested in porting to another RTOS or a heapless implementation will be a good time to design the portability strategy — we'll have to have documentation at that point of a clear list of the functions outside these libraries that the system must implement. And before that, we'll want to actually make Bristlemouth a proper library that can be included in other code.

**Review by towynlin — APPROVED** [review]

File: config_cbor_map_srv_reply_msg.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (victorsowa12) [reply config_cbor_map_srv_reply_msg.cpp:249]: Cool cool! I think this is definitely something we can revisit later. 

Sorry port was the wrong word choice. I meant more, if someone _includes_ `bm_serial` (and therefore `bm_common_messages` as a submodule of `bm_serial`) in their Arduino or Zephyr application processor code they would run into issues with the FreeRTOS functions. Just thoughts for future applications - no action needed at this moment!

**Pull request merged**